source/common/http/filter_manager.cc

```diff
@@ -1374,7 +1374,8 @@ bool ActiveStreamDecoderFilter::recreateStream(const ResponseHeaderMap* headers)
   // Because the filter's and the HCM view of if the stream has a body and if
   // the stream is complete may differ, re-check bytesReceived() to make sure
   // there was no body from the HCM's point of view.
-  if (!complete() || parent_.stream_info_.bytesReceived() != 0) {
+  if (!parent_.enableInternalRedirectsWithBody() &&
```

Comment:
I think we want
if (!complete() || (!parent_.enableInternalRedirectsWithBody() parent_.stream_info_.bytesReceived() != 0)
no?

Reply:
Ah you're right - the integration test still passed since the router has the `downstream_end_stream_` flag that _generally_ prevents us from reaching this point on incomplete requests but (as the above comment mentions) this seems to be an extra protection layer against incomplete requests.

I don't see any existing tests for this branch but I can try to put one together - this occurs when the router's `downstream_end_stream_` is `true` but the FM's `state_.local_complete_` is `false`.